samza-azure/src/main/java/org/apache/samza/system/eventhub/EventHubSystemDescriptor.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.system.eventhub;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.samza.operators.descriptors.base.system.SystemDescriptor;
+import org.apache.samza.serializers.Serde;
+import org.apache.samza.system.eventhub.producer.EventHubSystemProducer.PartitioningMethod;
+
+
+/**
+ * A descriptor for a Event Hubs system.
+ * <p>
+ * System properties configured using a descriptor override corresponding properties provided in configuration.
+ */
+public class EventHubSystemDescriptor extends SystemDescriptor<EventHubSystemDescriptor> {
+  private static final String FACTORY_CLASS_NAME = EventHubSystemFactory.class.getName();
+
+  private List<String> streamIds = new ArrayList<>();
+  private Optional<Integer> fetchRuntimeInfoTimeout = Optional.empty();
+  private Optional<Integer> numClientThreads = Optional.empty();
+  private Optional<Integer> consumerReceiveQueueSize = Optional.empty();
+  private Optional<Integer> consumerMaxEventCountPerPoll = Optional.empty();
+  private Optional<Integer> consumerPrefetchCount = Optional.empty();
+  private Optional<Boolean> producerEventhubsSendKey = Optional.empty();
+  private Optional<PartitioningMethod> producerEventhubsPartitioningMethod = Optional.empty();
+
+  /**
+   * Constructs a {@link SystemDescriptor} instance.
+   *  @param systemName name of this system
+   */
+  public EventHubSystemDescriptor(String systemName) {
+    super(systemName, FACTORY_CLASS_NAME, null, null);
+  }
+
+  public <StreamMessageType> EventHubInputDescriptor<StreamMessageType> getInputDescriptor(String streamId, String namespace,
```

Comment:
Need to add javadoc for public methods, since they don't override the base class methods anymore.